src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/ReferencedPackageModel.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Protocol.Model;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI.Models.Package
+{
+    internal class ReferencedPackageModel : PackageModel, IDeprecationCapable, IVulnerableCapable
+    {
+        private readonly IDeprecationCapable _deprecationCapability;
```

Comment:
I find it unusual to have something that's of that interface and have something that implements that same interface.

I see we're doing some delegation:

>         public IReadOnlyList<PackageVulnerabilityMetadataContextInfo>? Vulnerabilities => _vulnerableCapability.Vulnerabilities;

You can also consider a new type that extends from PackageModel that can handle this for you since this code is not duplicated. 

Every type seems to handle vulnerability, so that extra implementation could reduce the amount of code. 
Just an idea.

Reply:
Thanks for the feedback! I considered several alternatives but landed on this design. I like the flexibility this design offers. However, if an alternative design comes up, we can adjust very easily to that new design from here. We talked about this in our meeting and neither of us could come up with a better design yet, so we'll stick with this one and iterate on it if needed.